nemo/collections/asr/modules/audio_modules.py

```diff
@@ -718,6 +718,257 @@ def forward(self, input: torch.Tensor, input_length: torch.Tensor) -> Tuple[torc
         return masks, mask_length
 
 
+class MaskEstimatorGSS(NeuralModule):
+    """Estimate masks using guided source separation with a complex
+    angular Central Gaussian Mixture Model (cACGMM).
+
+    Notation is approximately following [1], where `gamma` denotes
+    the time-frequency mask, `alpha` denotes the mixture weights,
+    and `BM` denotes the shape matrix. Additionally, the provided
+    source activity is denoted as `activity`.
+
+    Args:
+        num_iterations: Number of iterations for the EM algorithm
+        eps: Small value for regularization
+        dtype: Data type for internal computations (default `torch.cdouble`)
+
+    References:
+        [1] Ito et al., Complex Angular Central Gaussian Mixture Model for Directional Statistics in Mask-Based Microphone Array Signal Processing, 2016
+        [2] Boeddeker et al., Front-End Processing for the CHiME-5 Dinner Party Scenario, 2018
+    """
+
+    def __init__(self, num_iterations: int = 3, eps: float = 1e-8, dtype: torch.dtype = torch.cdouble):
+        super().__init__()
+
+        if num_iterations <= 0:
+            raise ValueError(f'Number of iterations must be positive, got {num_iterations}')
+
+        # number of iterations for the EM algorithm
+        self.num_iterations = num_iterations
+
+        if eps <= 0:
+            raise ValueError(f'eps must be positive, got {eps}')
+
+        # small regularization constant
+        self.eps = eps
+
+        # internal calculations
+        if dtype not in [torch.cfloat, torch.cdouble]:
+            raise ValueError(f'Unsupported dtype {dtype}, expecting cfloat or cdouble')
+        self.dtype = dtype
+
+        logging.debug('Initialized %s', self.__class__.__name__)
+        logging.debug('\tnum_iterations: %s', self.num_iterations)
+        logging.debug('\teps:            %g', self.eps)
+        logging.debug('\tdtype:          %s', self.dtype)
+
+    def normalize(self, x: torch.Tensor, dim: int = -3) -> torch.Tensor:
```

Comment:
Is there a specific reason for making this -3 than 1 ? If batchsize=1 creates problem, this feels like a right choice.